src/core/searcher.rs

```diff
@@ -10,9 +9,62 @@ use crate::space_usage::SearcherSpaceUsage;
 use crate::store::StoreReader;
 use crate::DocAddress;
 use crate::Index;
+use crate::Opstamp;
+use crate::SegmentId;
+use crate::SegmentMeta;
+use crate::TrackedObject;
 
+use std::collections::BTreeMap;
 use std::{fmt, io};
 
+/// Identifies the index generation accessed by a [Searcher].
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct SearcherIndexGeneration {
```

Comment:
Shall we name this back to `SearcherGeneration`? When it was wrapping `segments` it was only concerned with index state, but with `generation_id` it also contains an identifier for a generation of `Searcher`.

Reply:
Yes I think that would be better.